File: src/main/java/dk/cngroup/wishlist/entity/Product.java
package dk.cngroup.wishlist.entity;

import jakarta.persistence.Entity;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;
import org.springframework.data.rest.core.annotation.Description;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Product extends AuditableEntity {
    @Description("Unique name of the item")
    @NotNull
    @Length(min = 3, max = 30)
    String code;

    public Product(String code) {
        this.code = code;
    }
}

File: src/main/kotlin/dk/cngroup/wishlist/entity/Product.kt
package dk.cngroup.wishlist.entity

import jakarta.persistence.Entity
import jakarta.validation.constraints.NotNull
import org.springframework.data.jpa.repository.JpaRepository

@Entity
class Product(
    @field:NotNull
    var code: String
) : AuditableEntity()

interface ProductRepository : JpaRepository<Product, Long>
